File: backend/src/main/java/com/adhamamr/passwordy/dto/AccountExportResponse.java
package com.adhamamr.passwordy.dto;

import java.time.Instant;
import java.util.List;

/**
 * A complete, machine-readable export of everything the service holds about a user (GDPR data
 * portability). Vault entries are returned <b>decrypted</b> so the export is actually usable for
 * migration — it is only ever produced for the authenticated owner over the (TLS) API.
 */
public record AccountExportResponse(
        ExportedAccount account,
        List<ExportedPassword> passwords,
        Instant exportedAt
) {
    /** The account's own profile fields (no password hash is ever exported). */
    public record ExportedAccount(
            String username,
            String email,
            boolean twoFactorEnabled,
            Instant createdAt
    ) {}

    /** One vault entry with its decrypted secret. */
    public record ExportedPassword(
            String label,
            String username,
            String password,
            String url,
            String notes,
            String category,
            boolean favorite,
            Instant createdAt,
            Instant updatedAt
    ) {}
}

File: frontend/app/src/main/java/com/adhamamr/passwordy/data/model/AccountExportResponse.kt
package com.adhamamr.passwordy.data.model

/**
 * Mirrors the backend GDPR export from `GET /api/account/export`. Timestamps arrive as ISO-8601
 * strings (Spring's default Instant serialization); kept as String since the export is written
 * straight back out to a file.
 */
data class AccountExportResponse(
    val account: ExportedAccount,
    val passwords: List<ExportedPassword>,
    val exportedAt: String?
)

data class ExportedAccount(
    val username: String,
    val email: String,
    val twoFactorEnabled: Boolean,
    val createdAt: String?
)

data class ExportedPassword(
    val label: String?,
    val username: String?,
    val password: String?,
    val url: String?,
    val notes: String?,
    val category: String?,
    val favorite: Boolean,
    val createdAt: String?,
    val updatedAt: String?
)
